mantle/platform/qemu.go

```diff
@@ -1198,19 +1207,34 @@ func (builder *QemuBuilder) addDiskImpl(disk *Disk, primary bool) error {
 			panic(fmt.Sprintf("Mantle doesn't know which bus type to use on %s", builder.architecture))
 		}
 
-		for i := 0; i < 2; i++ {
-			if i == 1 {
-				opts = strings.Replace(opts, "bootindex=1", "bootindex=2", -1)
+		if disk.MultiPathDisk {
+			// All these attributes are needed in order
+			// to trick multipath-tools that this is a "real" multipath device.
+			// Each disk is presented on its own controller.
+
+			for i := 0; i < 2; i++ {
+				if i == 1 {
+					opts = strings.Replace(opts, "bootindex=1", "bootindex=2", -1)
+				}
+				pID := fmt.Sprintf("mpath%d%d", builder.diskID, i)
+				scsiID := fmt.Sprintf("scsi_%s", pID)
+				builder.Append("-device", fmt.Sprintf("virtio-scsi-%s,id=%s", bus, scsiID))
+				builder.Append("-device",
+					fmt.Sprintf("scsi-hd,bus=%s.0,drive=%s,vendor=NVME,product=VirtualMultipath,wwn=%d%s",
```

Comment:
Note that `wwn=%d` is a hair misleading : qemu-exec will convert this integer value to it's base 16 representation, and the link will be created with that. 
So if  you pass `5G:channel=scsi,wwn=0011` you get `/dev/disk/by-id/scsi-300000000000000b`

A way around that would be 
```suggestion
					fmt.Sprintf("scsi-hd,bus=%s.0,drive=%s,vendor=NVME,product=VirtualMultipath,wwn=0x%d%s",
```
But then you force the base 10 number to be intepreted as base16. This works because base 10 is a subset of base 16 but that feels a bie weird to do, I'm curious to get a second opinion on that.

Reply:
YAML supports hex numbers, so I would leave it as is and we could write it in hex in the test's YAML metadata so it matches the symlink.